### PR TITLE
Demo UX improvements

### PR DESCRIPTION
**Pull request by Pessimistress**
- Use sliders for range parameters
- Allow demos to set minZoom and disable perspective mode
- Icon example use various sizes in cluster mode
- Line example use additive blending
- Reorganize examples, improve wording
- Perf for line and icon demos
- Default parameter tweaks

### CODE REVIEW

File: demo/src/components/demos/heatmap-demo.js [hunk 1]
```diff
@@ -15,29 +15,10 @@ export default class HeatmapDemo extends Component {
...
-      },
-      upperPercentile: {
-        displayName: 'Upper Percentile',
-        type: 'number',
-        value: 100,
-        step: 0.5,
-        min: 0,
-        max: 100
-      }
+      radius: {displayName: 'Radius', type: 'range', value: 2000, step: 500, min: 500, max: 20000},
```
Comment (heshan0131) [inline demo/src/components/demos/heatmap-demo.js]: Can you change radius to
`step: 100`

percentile to
`step: 0.1`

File: examples/screen-grid/app.js [hunk 2]
```diff
@@ -54,7 +54,6 @@ class Root extends Component {
       <MapGL
         {...viewport}
         mapStyle="mapbox://styles/mapbox/dark-v9"
-        perspectiveEnabled={true}
```
Comment (ibgreen) [inline examples/screen-grid/app.js]: It's a bit unfortunate to remove them here in examples, even if they are duplicated.

File: demo/src/components/demos/icon-demo.js [hunk 3]
```diff
@@ -101,17 +101,33 @@ export default class IconDemo extends Component {
...
+
     return (
-      <div className={`tooltip ${expanded ? 'interactive' : ''}`}
-           style={{left: x, top: y}}
-           onWheel={stopPropagation}
-           onMouseDown={stopPropagation}>
+      <div className="tooltip" style={{left: x, top: y}} >
         {
-          hoveredItems.map(([name, scenes]) => {
+          Object.keys(hoveredItems).slice(0, 20).map(name => {
```
Comment (heshan0131) [inline demo/src/components/demos/icon-demo.js]: nit, you can use => without a return
```
name => (<div key={name}><h5>{name}</h5></div>)
```

File: demo/src/reducers/viewport.js [hunk 4]
```diff
@@ -9,8 +9,11 @@ export default handleActions({
 
   UPDATE_MAP: (state, action) => {
     const {viewport} = action;
-    const maxZoom = viewport.maxZoom || state.maxZoom || 20;
-    if (viewport.zoom > maxZoom) {
+
+    const maxZoom = viewport.maxZoom === undefined ? state.maxZoom : viewport.maxZoom;
```
Comment (heshan0131) [inline demo/src/reducers/viewport.js]: maybe `Number.isFinit(viewport.maxZoom)` is safer here

File: examples/screen-grid/app.js (hunk 2 above)
Comment (Pessimistress) [inline examples/screen-grid/app.js]: I'll change it to `false` instead

**Pull request merged**